### DISCUSSION TITLE
Use strapi with Azure Compass DB (Core SQL)

### DISCUSSION THREAD

**Discussion by ghost** [strapi/strapi#7607] [Legacy Topics]
Hey, so i was wondering since the documentation mentions about using Azure Compass (MongoDB) is possible, if the same goes for the SQL version and what connector package i would have to use for that one?

I've seen that in NodeJS, you connect and query Azure Compass SQL using their own package. https://www.npmjs.com/package/@azure/cosmos

**Comment by derrickmehaffy** [comment]
Azure Cosmos DB is not a DB we officially support as it's not entirely MongoDB, It's a proprietary NoSQL that claims to be "MongoDB compatible" but in testing, it's not entirely and hence why we don't support it.

Currently we do not have any recommendations or guides to help in this case as we only officially support the databases and their versions listed on the README.md page. Among the ones that Azure offers that we would technically support it would be:

- Azure Database for PostgreSQL
- Azure Database for MySQL
- Azure Database for MariaDB

The most likely cause as to why Azure doesn't officially have a MongoDB service is due to the company behind MongoDB wanting to charge an outrageous cost to clouds to offer the service.

**Reply by ghost** [reply]
Thanks for the reply, but i was more specifically speaking about the SQL version of Compass, not the MongoDB.

Moreover, I've seen that Azure provides a serverless Microsoft SQL `mssql` service - is that one supported or compatible with some tweaks?

**Reply by derrickmehaffy** [reply]
MSSQL (Microsoft SQL Server) is not currently supported, this is outlined in this issue: https://github.com/strapi/strapi/issues/5791

For the moment we are not considering MSSQL, but we are leaving that there in case we feel we can support it in the future.

**Comment by brunoprietog** [comment]
@derrickmehaffy  In addition to the api of Mongo db, Azure Cosmos DB also offers an api to make queries in the form of SQL, would it be possible to implement this? Or do you have the same problems as with the Mongo DB api implemented in Cosmos DB? Thank you

**Reply by derrickmehaffy** [reply]
Same thing, it's not a typical MongoDB/SQL database, it is a proprietary database, and due to that it's not something we plan to support right now.
